### PR TITLE
[IMPL] TabGUI

### PR DESCRIPTION
**Pull request by Wolfsurge**
hopefully this isn't too bad :P

Here's what it looks like
tabgui

### CODE REVIEW

**Comment by master7720** [comment]
looks great

**Comment by Wolfsurge** [comment]
cool 👍

@linustouchtips pls merge

**Comment by Sxmurai** [comment]
Good job, I don't see any code changes from a quick glance. I'll let linus review and push into main.

File: src/main/java/cope/cosmos/client/features/modules/client/HUDModule.java [hunk 1]
```diff
@@ -268,6 +272,19 @@ public void onRender2D() {
...
+        if (tabGUI.getValue()) {
+            Cosmos.INSTANCE.getTabGUI().render();
+        }
+    }
+
+    @SubscribeEvent
+    public void onKeyInput(InputEvent.KeyInputEvent event) {
+        if (mc.currentScreen == null) {
+            if (tabGUI.getValue()) {
+                Cosmos.INSTANCE.getTabGUI().onKeyPress(event);
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/features/modules/client/HUDModule.java:285]: use getCosmos()

File: src/main/java/cope/cosmos/client/features/modules/client/HUDModule.java [hunk 2]
```diff
@@ -268,6 +272,19 @@ public void onRender2D() {
                 });
             }
         }
+
+        if (tabGUI.getValue()) {
+            Cosmos.INSTANCE.getTabGUI().render();
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/features/modules/client/HUDModule.java:277]: use getCosmos()

File: src/main/java/cope/cosmos/client/ui/tabgui/TabGUI.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import cope.cosmos.client.ui.tabgui.component.CategoryComponent;
+import net.minecraftforge.fml.common.gameevent.InputEvent;
+import org.lwjgl.input.Keyboard;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Wolfsurge
+ */
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/ui/tabgui/TabGUI.java:14]: date?

File: src/main/java/cope/cosmos/client/ui/tabgui/component/CategoryComponent.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        for (Module module : Cosmos.INSTANCE.getModuleManager().getModules(module -> module.getCategory() == category)) {
+            // If module name is longer than max width, set max width to that
+            if (FontUtil.getStringWidth(module.getName()) + 16 > maxWidth) {
+                maxWidth = FontUtil.getStringWidth(module.getName()) + 16;
+            }
+        }
+
+        // Add module components
+        float yOffset = 0;
+        for (Module module : Cosmos.INSTANCE.getModuleManager().getModules(module -> module.getCategory().equals(category))) {
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/ui/tabgui/component/CategoryComponent.java:67]: use getCosmos()

**Review by linustouchtips — CHANGES_REQUESTED** [review]
looks good, remember to also merge branch main

File: src/main/java/cope/cosmos/client/ui/tabgui/component/CategoryComponent.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+import cope.cosmos.util.string.StringFormatter;
+import net.minecraftforge.fml.common.gameevent.InputEvent;
+import org.lwjgl.input.Keyboard;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Wolfsurge
```
[the remark this answers is not in the document]
Comment (master7720) [reply src/main/java/cope/cosmos/client/ui/tabgui/component/CategoryComponent.java:20]: doesnt really need a date tbh

[the remark this answers is not in the document]
Comment (Wolfsurge) [reply src/main/java/cope/cosmos/client/ui/tabgui/component/CategoryComponent.java:20]: oh well i already fixed i

[the remark this answers is not in the document]
Comment (linustouchtips) [reply src/main/java/cope/cosmos/client/ui/tabgui/component/CategoryComponent.java:20]: > doesnt really need a date tbh

OMG STFU

**Pull request merged**